### PR TITLE
Drawing player/add common

### PR DESCRIPTION
**Pull request by learnforward2023**
## Why

Fixes: Drawing player/add common

## What

- [Create header file Common.h](https://github.com/learnforward2023/sky-birds-game/commit/73cd2e02c65aa973045836fa513444d896bdc9df)
- [🛠️ Move common variables to common.h](https://github.com/learnforward2023/sky-birds-game/commit/f05040548040c070525e7677364cf5293c509aea)
- [📁 Refactor directory structure](https://github.com/learnforward2023/sky-birds-game/commit/1fbb9e4dcfc87f1255464d5ac58dee8a8e392f1f)
- [Refactor code according to the BOT's suggestions](https://github.com/learnforward2023/sky-birds-game/pull/3/commits/274297e88e2ce4ed0dd56259a0becaaae5286598)

## Other Information

Website: https://studytogether.vn/ 🌟
Tiktok: [Lets Study Together](https://www.tiktok.com/@study2together) 🙏

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -1,19 +1,14 @@
-#include <SDL.h>
-#include <SDL_image.h>
-#include <string>
+#include "src/common/Common.h"
```
Comment (learnforward2023) [inline main.cpp:1]: Tại sao lại xoá hết mấy cái `#include` kia đi mà để lại mỗi `#include "src/common/Common.h"` 🤔

**Comment by learnforward2022** [comment]
Summary of Changes:

1. The image file `background.png` was moved and renamed. Previously, it was located under `images/`, but now it is under `assets/images/background/` and renamed to `primary.png`.

2. In `main.cpp`, includes for SDL2 (`<SDL.h>` and `<SDL_image.h>`) and `<string>` have been replaced with an include for `"src/common/Common.h"`.

3. Constants `GAME_SCREEN_WIDTH`, `GAME_SCREEN_HEIGHT`, and `GAME_TITLE` have been removed from `main.cpp` and are now declared in `Common.h`.

4. The function `renderBackground()` in `main.cpp` now uses the new path for `background.png`, which has been updated to reflect the new location and name of the background image file.

5. Two new files `Common.cpp` and `Common.h` have been created under `src/common/`. `Common.h` defines the removed constants `GAME_SCREEN_WIDTH`, `GAME_SCREEN_HEIGHT`, and `GAME_TITLE`, and includes necessary SDL headers and `string`.

Points to Note:

- By moving global constants into a `Common.h` file, they are now centralized, which is a good practice for maintainability.

- Grouping all common SDL includes into one common header file reduces redundancy in include statements across different files.

Helpful Tips and Tricks:

- When moving and renaming files like the background image, make sure to update all references to the file in your codebase, or better yet, consider using a resource manager class to handle file paths.

- A common header like `Common.h` could potentially grow very large and become unmanageable as more components are added. It's worth considering splitting it into more specialized headers if the project grows (e.g., `GraphicsSettings.h`, `GameSettings.h`, etc.).

- You can use `#pragma once` at the top of your header files to prevent multiple inclusions instead of traditional include guards, which can be simpler and less error-prone.

Other Solutions:

- For larger projects, you might consider using a build system like CMake, which can help manage file paths and dependencies more efficiently.

- Instead of hardcoding paths, you could use a configuration system that reads paths from a configuration file or environment variables, making your application more flexible and easier to deploy.

Advice to Improve the Code:

- Make sure your code is properly version-controlled and that moving files around in the file system is accurately reflected in your version control history.

- Use a consistent coding style and naming convention for files and directories. For example, if you're using `snake_case` or `camelCase` for naming, stick to one.

- Avoid global static variables in header files, as they could lead to multiple definitions across different compilation units, violating the One Definition Rule (ODR). It's usually better to define them in a `.cpp` file.

- Implement proper error handling for file operations. For instance, when loading images or other assets, make sure you check if the file actually exists and is accessible, and provide a fallback or an error message if it's not.

- Consider building a more robust asset loading and management system to handle changes and lifecycle of assets as your project grows.

#### This comment is created by OpenAI API. 👍
Website: [https://studytogether.vn](https://studytogether.vn) 🌟
Tiktok: [Lets Study Together](https://www.tiktok.com/@study2together) 🙏

**Comment by learnforward2023** [comment]
> You can use #pragma once at the top of your header files to prevent multiple inclusions instead of traditional include guards, which can be simpler and less error-prone.

Oh, Good 🙇‍♂️ I already used
```c
// header.h
// Demonstration of the #include guard idiom.
// Note that the defined symbol can be arbitrary.
#ifndef HEADER_H_     // equivalently, #if !defined HEADER_H_
#define HEADER_H_

// Code placed here is included only once per translation unit

#endif // HEADER_H_
```

----------

> Avoid global static variables in header files, as they could lead to multiple definitions across different compilation units, violating the One Definition Rule (ODR). It's usually better to define them in a .cpp file.

👍 Great!

Fixes: [Refactor code according to the BOT's suggestions](https://github.com/learnforward2023/sky-birds-game/pull/3/commits/274297e88e2ce4ed0dd56259a0becaaae5286598)

File: src/common/Common.h [hunk 2]
```diff
@@ -10,8 +10,8 @@
 #include <SDL_image.h>
 #include <string>
 
-static const int GAME_SCREEN_WIDTH = 1400;
-static const int GAME_SCREEN_HEIGHT = 740;
-static const char* GAME_TITLE = "Study Together - Sky Birds Game";
+extern const int GAME_SCREEN_WIDTH;
+extern const int GAME_SCREEN_HEIGHT;
+extern const char* GAME_TITLE;
```
Comment (learnforward2023) [inline src/common/Common.h:15]: - `extern` là gì, sử dụng trong trường hợp nào?
- Ở đây có sự nhầm lẫn về cách sử dụng của static, hãy tìm ra sự nhầm lẫn đó để khắc phục trong tương lai 🙏

File: src/common/Common.cpp [hunk 3]
```diff
@@ -0,0 +1,7 @@
+//
+// Created by Study Together on 2024/02/11.
+//
+
+const int GAME_SCREEN_WIDTH = 1400;
+const int GAME_SCREEN_HEIGHT = 740;
+const char* GAME_TITLE = "Study Together - Sky Birds Game";
```
Comment (learnforward2023) [inline src/common/Common.cpp:7]: - Tại sao ở đây không cần `#include "Common.h"` mà vẫn sử dụng được nhỉ? 🤔

**Pull request merged**